src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -164,25 +176,46 @@ public static Person createDeletePerson(Person personToEdit, DeletePersonDescrip
         Appointment updatedAppointment = personToEdit.getAppointment().isPresent() ? personToEdit.getAppointment().get()
                 : null;
 
-        if (deletePersonDescriptor.shouldDeleteMedicalHistory()) {
-            updatedMedicalHistories = new HashSet<MedicalHistory>();
+        if (deletePersonDescriptor.shouldDeleteAppointment()) {
+            if (updatedAppointment == null) {
+                throw new CommandException(MESSAGE_NO_APPOINTMENT_TO_DELETE);
+            } else {
+                updatedAppointment = null;
+            }
         }
 
-        if (deletePersonDescriptor.shouldDeleteAppointment()) {
-            updatedAppointment = null;
+        if (deletePersonDescriptor.shouldDeleteMedicalHistory()) {
+            Set<MedicalHistory> medicalHistoriesToDelete = deletePersonDescriptor.getMedicalHistories();
+            Set<MedicalHistory> medicalHistoriesToKeep = new HashSet<>();
+            if (updatedMedicalHistories.isEmpty()) {
+                throw new CommandException(MESSAGE_NO_MEDICAL_HISTORY_TO_DELETE);
+            } else if (medicalHistoriesToDelete == null || medicalHistoriesToDelete.isEmpty()) {
+                updatedMedicalHistories = new HashSet<>();
+            } else {
+                if (!updatedMedicalHistories.containsAll(medicalHistoriesToDelete)) {
+                    throw new CommandException(MESSAGE_INVALID_MEDICAL_HISTORY);
+                }
+                for (MedicalHistory medicalHistory : updatedMedicalHistories) {
+                    if (!medicalHistoriesToDelete.contains(medicalHistory)) {
+                        medicalHistoriesToKeep.add(medicalHistory);
+                    }
+                }
+                updatedMedicalHistories = medicalHistoriesToKeep;
+            }
```

Comment:
I like this implementation to create a temporary new HashSet to store the result before updating the result of the deleting of the medical history.